[gatsby-source-wordpress] Refactored featured_media map for deep nodes

sebastienfi opened the pull request:
* Featured medias could be nested at any level
* The most certain way to have photos works with this version on the plugin is to either name the field featured_media or include the image as Post Object.
* Fixed bug where in some case the old featured_media field was not deleted.

Tested with using-wordpress OK.

sebastienfi commented:
Funny enough pieh has been working on something similar with this PR https://github.com/gatsbyjs/gatsby/pull/2646

I believe my PR is more generic as it allows for featured_medias to be placed anywhere in the JSON tree, let's say for example, in a linked Post Object.

KyleAMathews commented:
haha, brilliant minds…

I'll let you make the call on which one to merge.

pieh commented:
It's nobrainer that You should take Your code, as most likely You will maintain it ;) Just check my comment https://github.com/gatsbyjs/gatsby/pull/2646#issuecomment-339847802 and I think You can close my PR and focus on this one.

sebastienfi commented:
* Most of the credits goes to @pieh for a better solution that solved the same problems and widely improved code simplicity. https://github.com/gatsbyjs/gatsby/pull/2646
* Made the Media lookup begin at JSON tree root instead of keeping this for ACF Field only. Changed constants names accordingly.
* Making thise more generic allorws Custom Post Types to benefit from this improvements and yet unseen objects shapes. 
* Incorporated fix on bug mentionned here https://github.com/gatsbyjs/gatsby/pull/2646#commitcomment-25218740
* May solve this https://github.com/gatsbyjs/gatsby/issues/2587, this https://github.com/gatsbyjs/gatsby/issues/2492, this https://github.com/gatsbyjs/gatsby/issues/2328
